annotations/views.py

```diff
@@ -149,7 +153,45 @@ def get_alignments(self):
         return Alignment.objects.select_related('original_fragment', 'translated_fragment')
 
 
-class AnnotationUpdateMixin(AnnotationMixin, CheckOwnerOrStaff):
+class RevisionWithCommentMixin(RevisionMixin):
+    revision_manage_manually = True
+
+    def form_valid(self, form):
+        result = super().form_valid(form)
+        change = construct_change_message(form, None, False)
+        if change:
+            reversion.add_to_revision(self.object)
+            reversion.set_comment(self.format_change_comment(change, form.cleaned_data))
+
+        return result
+
+    def format_change_for_field(self, field, value):
+        if isinstance(value, QuerySet):
+            value = ', '.join(map(str, value))
+        return '{} to "{}"'.format(field, value)
+
+    def format_change_comment(self, change, values):
+        change = change[0]
+        if 'changed' in change and 'fields' in change['changed']:
+            fields = change['changed']['fields']
+            parts = [self.format_change_for_field(field, values[field])
+                     for field in fields]
+            return 'Changed {}'.format(', '.join(parts))
+        return ''
```

Comment:
I find this a bit weird. When does this happen?

Reply:
I'm not sure why it's here. Could be related to earlier attempts during development. Would you rather have an error raised?

Reply:
Let's just add a TODO comment that this should not happen :-) I might investigate later.